cluster_experiments/experiment_analysis.py

```diff
@@ -539,3 +539,108 @@ def from_config(cls, config):
+    from cluster_experiments.experiment_analysis import MLMExperimentAnalysis
+    import pandas as pd
+
+    df = pd.DataFrame({
+        'x': [1, 2, 3, 0, 0, 1],
+        'treatment': ["A"] * 3 + ["B"] * 3,
+        'cluster': [1] * 6,
+    })
+
+    MLMExperimentAnalysis(
+        cluster_cols=['cluster'],
+        target_col='x',
+    ).get_pvalue(df)
+    ```
+    """
+
+    def __init__(
+        self,
+        cluster_cols: List[str],
+        target_col: str = "target",
+        treatment_col: str = "treatment",
+        treatment: str = "B",
+        covariates: Optional[List[str]] = None,
+    ):
+        super().__init__(
+            target_col=target_col,
+            treatment_col=treatment_col,
+            cluster_cols=cluster_cols,
+            treatment=treatment,
+            covariates=covariates,
+        )
+        # the specification with C() ensures consistent ordering of treatment and non-treatment in the model
+        self.mlm_treatment_spec = (
+            f"C({self.treatment_col}, Treatment(reference='{self.treatment}'))"
+        )
+        self.mlm_treatment_label = None
+        self.formula = f"{self.target_col} ~ {self.mlm_treatment_spec}"
+        if self.covariates:
+            self.formula = f"{self.formula} + {' + '.join(self.covariates)}"
+
+        self.re_formula = None
+        self.vc_formula = None
+
+    def fit_mlm(self, df: pd.DataFrame) -> sm.MixedLM:
+        """Returns the fitted MLM model"""
+        self.mlm_treatment_label = (
+            f"{self.mlm_treatment_spec}[T.{self._get_non_treatment_label(df)}]"
+        )
+        return sm.MixedLM.from_formula(
+            formula=self.formula,
+            data=df,
+            groups=self._get_cluster_column(df),
+            re_formula=self.re_formula,
+            vc_formula=self.vc_formula,
+        ).fit()
+
+    def _get_non_treatment_label(self, df):
+        return (set(df[self.treatment_col]) - set([self.treatment])).pop()
+
+    def analysis_pvalue(self, df: pd.DataFrame, verbose: bool = False) -> float:
+        """Returns the p-value of the analysis
+        Arguments:
+            df: dataframe containing the data to analyze
+            verbose (Optional): bool, prints the regression summary if True
+        """
+        results_mlm = self.fit_mlm(df)
+        if verbose:
+            print(results_mlm.summary())
+        return results_mlm.pvalues[self.mlm_treatment_label]
+
+    def analysis_point_estimate(self, df: pd.DataFrame, verbose: bool = False) -> float:
+        """Returns the point estimate of the analysis
+        Arguments:
+            df: dataframe containing the data to analyze
+            verbose (Optional): bool, prints the regression summary if True
+        """
+        results_mlm = self.fit_mlm(df)
+        # negative sign, as we use the treatment as reference (instead of the baseline)
+        return -results_mlm.params[self.mlm_treatment_label]
+
+    def get_pvalue(self, df: pd.DataFrame) -> float:
```

Comment:
why do you need to override this?

Reply:
I could also have overridden `_get_binary_treatment` to just return the input dataframe, but I felt that having a method called `_get_binary_treatment` that's just the identity function is a bit misleading and may cause problems elsewhere... Would you prefer this option?

Reply:
what's the issue of not overriding any of the 2?